server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -254,10 +257,53 @@ static private void downloadOnce(TranslogTransferManager translogTransferManager
                 location.resolve(Translog.getCommitCheckpointFileName(translogMetadata.getGeneration())),
                 location.resolve(Translog.CHECKPOINT_FILE_NAME)
             );
+        } else {
+            // When code flow reaches this block, it means we don't have any translog files uploaded to remote store.
+            // If local filesystem contains empty translog or no translog, we don't do anything.
+            // If local filesystem contains non-empty translog, we clean up these files and create empty translog.
+            logger.debug("No translog files found on remote, checking local filesystem for cleanup");
+            if (FileSystemUtils.exists(location.resolve(CHECKPOINT_FILE_NAME))) {
```

Comment:
Is there any possibility of checkpoint file not existing, but translog file being there?

Reply:
tlog files without checkpoint should be cleaned up. The issue is with tlog + ckp files which can be read and considered live.